lib/devise-security/validators/complexity_validator.rb

```diff
@@ -0,0 +1,28 @@
+module DeviseSecurity
+  # String complexity validator
+  # Validates complexity of e.g. passwords
+  # Options:
+  # - digit:  minimum number of digits in the validated string
+  # - lower:  minimum number of lower-case letters in the validated string
+  # - symbol: minimum number of punctuation characters or symbols in the validated string
+  # - upper:  minimum number of upper-case letters in the validated string
+  class ComplexityValidator < ActiveModel::EachValidator
+    PATTERNS = {
+      digit: /\p{Digit}/,
```

Comment:
I think it's likely that some people will try to use `digits: 2` or `symbols: 2` in the configuration, which is silently fail to do the right thing.  Consider either making them aliases, or checking the keys to make sure they match the set that can be used.